src/main/java/org/folio/marc4ld/service/ld2marc/field/impl/Bibframe2MarcFieldRuleImpl.java

```diff
@@ -0,0 +1,139 @@
+package org.folio.marc4ld.service.ld2marc.field.impl;
+
+import static java.util.Objects.isNull;
+import static java.util.Optional.ofNullable;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.folio.ld.dictionary.PredicateDictionary;
+import org.folio.ld.dictionary.model.Resource;
+import org.folio.marc4ld.configuration.property.Marc4BibframeRules;
+import org.folio.marc4ld.service.condition.ConditionChecker;
+import org.folio.marc4ld.service.dictionary.DictionaryProcessor;
+import org.folio.marc4ld.service.ld2marc.field.Bibframe2MarcFieldRule;
+import org.folio.marc4ld.service.ld2marc.field.ControlFieldRule;
+import org.folio.marc4ld.service.ld2marc.field.IndicatorRule;
+import org.folio.marc4ld.service.ld2marc.field.SubFieldRule;
+import org.folio.marc4ld.service.ld2marc.field.param.ControlFieldParameter;
+import org.folio.marc4ld.service.ld2marc.field.param.SubFieldParameter;
+
+public class Bibframe2MarcFieldRuleImpl implements Bibframe2MarcFieldRule {
+
+  private static final String CONTROL_FIELD_PREFIX = "00";
+  private final IndicatorRule indProperty1;
+  private final IndicatorRule indProperty2;
+  private final String tag;
+  private final Marc4BibframeRules.FieldRule fieldRule;
+  private final Collection<ControlFieldRule> controlFieldRules;
+  private final Collection<SubFieldRule> subFieldRules;
+  private final ConditionChecker conditionChecker;
+
+  public Bibframe2MarcFieldRuleImpl(
+    String tag,
+    Marc4BibframeRules.FieldRule fieldRule,
+    ConditionChecker conditionChecker,
+    DictionaryProcessor dictionaryProcessor
+  ) {
+    this.tag = tag;
+    this.fieldRule = fieldRule;
+    this.conditionChecker = conditionChecker;
+    this.controlFieldRules = mapToControlFieldRules(dictionaryProcessor);
+    this.subFieldRules = mapToSubFieldRules();
+    this.indProperty1 = mapToIndicator(
+      Marc4BibframeRules.FieldRule::getInd1,
+      Marc4BibframeRules.Marc2ldCondition::getInd1
+    );
+    this.indProperty2 = mapToIndicator(
+      Marc4BibframeRules.FieldRule::getInd2,
+      Marc4BibframeRules.Marc2ldCondition::getInd2
+    );
+  }
+
+  @Override
+  public boolean isSuitable(Resource resource, PredicateDictionary predicate) {
+    return Objects.equals(fieldRule.getTypes(), resource.getTypeNames())
```

Comment:
Future enhancement that we need to implement -

Once we add "Lookup" feature in Marva UI, a work's "subject" could point to either one of the following 2 resources
  >> A resource with types `Concept`, `Person`
  >> A resource with type just `Person`

In both cases, we have to transform the resource back to marc field `600`. 
Due to the check `Objects.equals(fieldRule.getTypes(), resource.getTypeNames())`, marc 600 will not be created for subjects with type just `Person`. Something to fix in future.

@PBobylev  @aleksei-pronichev  @askhat-abishev

Reply:
Great note, move it to the chat